packages/grafana-data/src/dataframe/processDataFrame.ts

```diff
@@ -251,6 +251,10 @@ export function guessFieldTypeForField(field: Field): FieldType | undefined {
  * @param guessDefined Whether to guess types of fields with already defined types
  */
 export const guessFieldTypes = (series: DataFrame, guessDefined = false): DataFrame => {
+  if (!series.fields.length) {
```

Comment:
Is this if statement necessary? If the dataframe `series` has no fields, then the for loop at 258 should just be skipped and then the series will be returned at line 277 I think

Reply:
Good catch, this was necessary before since fields were undefined but now it's an empty array.